code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/message/views/MessageFrame.kt

```diff
@@ -58,22 +67,35 @@ internal fun MessageFrame(
     onCreated: (WebView) -> Unit,
     onDisposed: () -> Unit
 ) {
-    val currentConfiguration = LocalConfiguration.current
+    // The current context is the activity that is hosting the message. We can safely cast it to
+    // an Activity because this composable is always used within the context of the activity in
+    // the current implementation of UIService
+    val currentActivity = LocalContext.current.findActivity() ?: run {
+        onDisposed()
+        Log.debug(ServiceConstants.LOG_TAG, "MessageFrame", "Unable to get the current activity. Dismissing the message.")
+        return
+    }
+
+    // We want to calculate the height and width of the message based on the host activity's
+    // content view. This is because the message is displayed on top of the content view and
+    // we want to ensure that the message is displayed relative to the activity's size.
+    // This allows the message to be displayed correctly in split screen mode and other
+    // multi-window modes.
+    val contentView = currentActivity.findViewById<View>(android.R.id.content)
+    val contentHeightDp = with(LocalDensity.current) { contentView.height.toDp() }
+    val contentWidthDp = with(LocalDensity.current) { contentView.width.toDp() }
```

Comment:
There are a bunch of tests in `MessageScreenTests` which assert for the IAM height and width based on the screen height and width. Could you please check if those need to be fixed?